backend/features/github/github.service.js

```diff
@@ -0,0 +1,47 @@
+async function githubFetch(path, params = {}) {
+  const query = Object.keys(params).length
+    ? "?" + new URLSearchParams(params).toString()
+    : "";
+
+  const url = `https://api.github.com${path}${query}`;
+
+  const res = await fetch(url, {
+    headers: {
+      Authorization: `Bearer ${process.env.GITHUB_TOKEN}`,
+      "User-Agent": "SmartBlog-App",
+    },
+  });
+
+  if (!res.ok) {
+    const errData = await res.json().catch(() => ({}));
+    throw new Error(errData.message || `GitHub API error: ${res.status}`);
```

Comment:
json 변환이 안되는 에러도 먹히지 않게 처리해주면 좋겠네요 ㅎㅎ

```suggestion
    try {
        const errData = await res.json();
        throw new Error(errData.message || `GitHub API error: ${res.status}`);
    } catch {
        throw new Error('parse error~~');
    }
```